allennlp/models/reading_comprehension/bidaf.py

```diff
@@ -332,7 +333,7 @@ def from_params(cls, vocab: Vocabulary, params: Params) -> 'BidirectionalAttenti
         text_field_embedder = TextFieldEmbedder.from_params(vocab, embedder_params)
         num_highway_layers = params.pop_int("num_highway_layers")
         phrase_layer = Seq2SeqEncoder.from_params(params.pop("phrase_layer"))
-        similarity_function = SimilarityFunction.from_params(params.pop("similarity_function"))
+        matrix_attention = MatrixAttentionFactory.from_params(params.pop("similarity_function"))
```

Comment:
You don't need this factory at all, you just call `MatrixAttention.from_params()`.